src/main/java/seedu/rc4hdb/logic/parser/commandparsers/CommandParser.java

```diff
@@ -1,13 +1,12 @@
 package seedu.rc4hdb.logic.parser.commandparsers;
 
 import seedu.rc4hdb.logic.commands.Command;
-import seedu.rc4hdb.logic.parser.Parser;
 import seedu.rc4hdb.logic.parser.exceptions.ParseException;
 
 /**
  * Represents a Parser that is able to parse user input into a {@code Command} of type {@code T}.
  */
-public interface CommandParser<T extends Command> extends Parser<T> {
+public interface CommandParser<T extends Command> {
```

Comment:
Since the generic `Parser` interface is a functional interface currently used *only* for returning a `Command` from user input, it is good that we bound the type parameter by `Command` here. Removing `Parser.java` also avoids confusion for future developers in deciding which interface to implement for their methods.

However, do keep in mind that we might need to re-add `Parser.java` if we ever need to return other objects after parsing. 😄